classes/helper.php

```diff
@@ -192,12 +207,35 @@ private static function regular_expression_search(string $search, string $table,
      * Perform a search on a table and column.
      *
      * @param string $search The text to search for.
+     * @param string $table The table to search.
+     * @param array $columns The columns to search.
      * @param bool $regex Whether to use regular expression search.
-     * @param string $tables A comma separated list of tables and columns to search.
      * @param int $limit The maximum number of results to return.
      * @return array
+     * @throws moodle_exception
      */
-    public static function search(string $search, bool $regex = false, string $tables = '', int $limit = 0): array {
+    public static function search(string $search, string $table, array $columns, bool $regex = false, int $limit = 0): array {
+        // Perform the search for each table and column.
+        $results = [];
+        foreach ($columns as $column) {
+            // Perform the search on this column.
+            if ($regex) {
+                $results = array_merge($results, self::regular_expression_search($search, $table, $column, $limit));
```

Comment:
this is going to be one of the main bottle necks, we never want to search for stuff and put the results into memory, of if we do it needs to be small batches or row by row. Instead we should pass in the file handler we are writing to and just stream the results to that file, while at the same time printing to stdout what table and column we are working on

Reply:
I have removed this function, the cli now call search/regex  for each row